cli/internal/cmd/trackingplan/apply/apply.go

```diff
@@ -39,21 +42,25 @@ func NewCmdTPApply() *cobra.Command {
 			$ rudder-cli tp apply --loc </path/to/dir or file> --dry-run
 		`),
 		PreRunE: func(cmd *cobra.Command, args []string) error {
+			fmt.Println()
+
 			// Here we might need to do validate
 			localcatalog, err = readCatalog(catalogDir)
 			if err != nil {
 				return fmt.Errorf("reading catalog failed in pre-step: %w", err)
 			}
 
-			if err := validate.ValidateCatalog(validate.DefaultValidators(), localcatalog); err != nil {
-				return fmt.Errorf("validating catalog: %w", err)
+			if errs := validate.NewCatalogValidator().Validate(localcatalog); errs != nil {
+				fmt.Println("catalog errors:")
+				fmt.Println(errs.Error())
+
+				return errors.New(emoji.Sprintf("catalog invalid :x:"))
```

Comment:
nit: let's move emojis at the start of the line. Personally, I would also consider not adding the emoji in the error itself, and use it as part of the validation report that's printed before. The reason is that this error is handled in a generic way by the framework and already prints an "Error" label.